tests/testthat/test-rng.R

```diff
@@ -1178,3 +1178,75 @@ test_that("numbers on different streams behave as expected", {
   expect_equal(res[, 2],
                dust_rng$new(1, seed = 1)$jump()$hypergeometric(10, m, n, k))
 })
+
+
+test_that("gamma for a = 1 is the same as exponential", {
+  rng1 <- dust_rng$new(seed = 1L)
+  rng2 <- dust_rng$new(seed = 1L)
+
+  n <- 10
+  b <- 3
+
+  gamma <- rng1$gamma(n, 1, b)
+  exp <- rng2$exponential(n, 1 / b)
+
+  expect_equal(gamma, exp)
+})
+
+
+test_that("can draw gamma random numbers", {
+  ## a > 1
+  a <- 5
+  b <- 3
+  n <- 10000000
+
+  ans1 <- dust_rng$new(1)$gamma(n, a, b)
+  ans2 <- dust_rng$new(1)$gamma(n, a, b)
+  expect_identical(ans1, ans2)
+
+  expect_equal(mean(ans1), a * b, tolerance = 1e-3)
+  expect_equal(var(ans1), a * b^2, tolerance = 1e-3)
+
+  ## a < 1
+  a <- 0.5
+
+  ans3 <- dust_rng$new(1)$gamma(n, a, b)
+  ans4 <- dust_rng$new(1)$gamma(n, a, b)
+  expect_identical(ans3, ans4)
+
+  expect_equal(mean(ans3), a * b, tolerance = 1e-3)
+  expect_equal(var(ans3), a * b^2, tolerance = 1e-3)
+})
+
+
+test_that("deterministic gamma returns mean", {
+  n_reps <- 10
+  a <- as.numeric(sample(10, n_reps, replace = TRUE))
+  b <- as.numeric(sample(10, n_reps, replace = TRUE))
+
+  rng_f <- dust_rng$new(1, real_type = "float", deterministic = TRUE)
+  rng_d <- dust_rng$new(1, real_type = "double", deterministic = TRUE)
+  state_f <- rng_f$state()
+  state_d <- rng_d$state()
+
+  expect_equal(rng_f$gamma(n_reps, a, b), a * b,
+               tolerance = 1e-6)
+  expect_equal(rng_d$gamma(n_reps, a, b), a * b)
+
+  expect_equal(rng_f$state(), state_f)
+  expect_equal(rng_d$state(), state_d)
+})
```

Comment:
add a test that with a == 1 you get _exactly_ the same stream as you'd expect from exponential?

Reply:
this one does that: https://github.com/mrc-ide/dust/pull/373/files#diff-957923b771229f7f58143ed5a4d8b30a42219ad7ec403e0bfc7221ee7c298fe6R1183

Reply:
sorry, totally missed that